### PR TITLE
[KernelGen][Nvidia] Add is_nonzero operator with Triton kernel

### PR DESCRIPTION
**Pull request by LoserCheems**
## Summary
Adds a Triton kernel for `is_nonzero`. Returns True if the single-element input tensor is not equal to zero. Uses `pointwise_dynamic` with `ALWAYS_BOOL` promotion.

## Testing
- Parametrized tests over `FLOAT_DTYPES + INT_DTYPES + BOOL_TYPES`
- Tests both non-zero and zero values, plus multi-element exception
- Validated against PyTorch reference on device side
- Tested on: Nvidia, Tianshu, Muxi, Ascend, Hygon

## Performance
Test command: `pytest benchmark/test_is_nonzero.py --level core` (NVIDIA H20)

**float16:**

| Shape | Torch (ms) | Gems (ms) | Speedup |
|---|---|---|---|
| [1] | 0.022 | 0.022 | 1.00x |
| **Arithmetic Mean** | — | — | **1.00x** |

**float32:**

| Shape | Torch (ms) | Gems (ms) | Speedup |
|---|---|---|---|
| [1] | 0.022 | 0.022 | 1.01x |
| **Arithmetic Mean** | — | — | **1.01x** |

**bfloat16:**

| Shape | Torch (ms) | Gems (ms) | Speedup |
|---|---|---|---|
| [1] | 0.022 | 0.022 | 1.00x |
| **Arithmetic Mean** | — | — | **1.00x** |

## Multi-backend Testing
| Backend | Accuracy Test | Benchmark | Speedup (mean) | Notes |
|---|---|---|---|---|
| Nvidia (H20) | PASS | PASS (15 cases, --level core) | 1.00x | Primary |
| Tianshu | PASS | PASS (36 cases) | 1.00x | — |
| Muxi | PASS | PASS (36 cases) | 1.00x | — |
| Ascend | PASS | FAIL | — | Benchmark failed |
| Hygon | PASS | PASS (36 cases) | 1.00x | — |

## Files Changed
- `src/flag_gems/ops/is_nonzero.py`: Triton kernel implementation
- `tests/test_is_nonzero.py`: Accuracy test
- `benchmark/test_is_nonzero.py`: Performance benchmark
- `src/flag_gems/ops/__init__.py`: Register import and `__all__`
- `src/flag_gems/__init__.py`: Register to `_FULL_CONFIG`
- `conf/operators.yaml`: Add operator entry (kind: Math, stage: alpha 5.1)

### CODE REVIEW

File: benchmark/test_is_nonzero.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+        # is_nonzero only accepts single-element tensors
+        yield torch.tensor([1], dtype=dtype, device=device)
+
+    def torch_is_nonzero(x):
+        return torch.is_nonzero(x)
+
+    bench = base.GenericBenchmark(
+        input_fn=is_nonzero_input_fn,
+        op_name="is_nonzero",
+        torch_op=torch_is_nonzero,
```
Comment (w1120029931-bit) [inline benchmark/test_is_nonzero.py]: Do not encapsulate, directly call torch.op

File: tests/test_is_nonzero.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+        res_out = torch.is_nonzero(inp)
+    ref_out = torch.is_nonzero(ref_inp)
+    assert res_out == ref_out, f"Expected {ref_out}, got {res_out}"
+
+    # Test zero values
+    inp_zero = torch.tensor([0], dtype=dtype, device=flag_gems.device)
+    ref_inp_zero = utils.to_reference(inp_zero)
+    with flag_gems.use_gems():
+        res_out_zero = torch.is_nonzero(inp_zero)
+    ref_out_zero = torch.is_nonzero(ref_inp_zero)
```
Comment (w1120029931-bit) [inline tests/test_is_nonzero.py]: ref_out_zero should be before res_out_zero

**Review by Dongxu-H — COMMENTED** [review]
## 代码审查：is_nonzero

感谢实现 `is_nonzero` 算子！以下是审查意见：

### ✅ 优点
1. 实现简洁明了
2. 正确处理标量输入
3. 添加了测试覆盖

### ⚠️ 需要关注的问题

**1. 输入形状限制**
`is_nonzero` 只接受标量输入（0-d tensor）。代码中是否有验证？
- 如果输入是多维 tensor，应该如何处理？
- 建议添加输入形状验证

**2. 返回类型**
返回值应该是标量 bool tensor。请确认：
- 返回类型是 `torch.Tensor` 而不是 Python `bool`
- dtype 是 `torch.bool`

**3. 特殊值处理**
需要确认以下情况的行为：
- `is_nonzero(0)` → False
- `is_nonzero(0.0)` → False
- `is_nonzero(nan)` → True（NaN 不等于 0）
- `is_nonzero(inf)` → True

**4. 与 PyTorch 行为一致性**
建议确保与 `torch.is_nonzero()` 完全一致：
```python
torch.is_nonzero(torch.tensor(0))  # False
torch.is_nonzero(torch.tensor([0]))  # RuntimeError
```

### 💡 建议
- 添加输入形状验证
- 添加更多边界情况测试（NaN, Inf, 负零等）
- 确认与 PyTorch 原生行为完全一致

实现整体正确，主要建议加强输入验证和边界测试。

**Comment by LoserCheems** [comment]
Thanks for the review! Fixed:

1. **Benchmark**: Removed the wrapper function, now directly passes `torch.is_nonzero` to `torch_op`.
2. **Test ordering**: Moved `ref_out_zero` before `res_out_zero` (reference computed first, then gems result).
3. **Rebased** on latest upstream/master, conflicts resolved.
4. **Stage** updated to `alpha: '5.4'`.

Squashed into a single commit.

**Review by Dongxu-H — COMMENTED** [review]
LGTM

**Review by Dongxu-H — APPROVED** [review]

**Review by Caeruleann — APPROVED** [review]
